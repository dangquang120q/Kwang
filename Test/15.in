69
22
573527939 109819568 301613663 11875701 54290099 558631374 426644218 839346354 961680563 657311748 204838304 237208712 398085976 869971784 102319050 839962543 298623394 726208243 493844497 675879873 521834541 560661375 
36
520742885 636738840 415560075 25149591 173776695 150142068 624592822 384176698 100290114 220765632 164662164 228592447 997616299 586407959 349931684 198591434 880737046 35994869 252384910 663659210 762401680 989940719 690639404 936073989 433154746 297131618 545260086 580009288 667111150 322339807 817372903 121507837 136312748 791773378 51446684 291302859 
27
754216630 247983475 455787361 452122462 203286848 119907402 77079844 69954739 958650296 385575938 222492565 511403850 67061822 508489182 199843581 42676461 330780261 598666972 33904494 673743610 866712398 366271216 861015306 26579924 841758672 124396245 158285578 
30
105203062 831347851 776206183 338226939 675532270 24936094 589035785 864135850 796127558 280607509 493452566 603047292 180841061 270822462 316838266 327597688 35306157 563227739 501059198 298747558 381071002 580394312 111656370 198209454 885498065 14773846 19493091 338207476 61531843 817728650 
43
190844273 468206196 601053131 943543588 735538453 352488752 740547082 170141282 553125877 330169674 899192366 53285297 142035830 890341071 153661668 362368748 200732788 517591091 312668533 991224817 138774532 156329371 667668654 999833440 611778225 661525182 68073809 436023532 676613498 864072462 80511260 673571054 293567796 631359072 612033432 967395789 246583858 626158944 189295116 469821887 661004556 616330741 764726665 
75
854435493 813113883 292127864 837459662 744930485 502393761 572511045 281710422 284104180 378642631 571801806 232268375 274898086 759767014 254273312 865654187 516495373 424758084 832630699 666709376 512560996 718645809 536257130 702307650 944351861 111091278 998821452 40815676 636042535 581622782 825794002 496424665 992252981 62857386 27818070 126628792 831396855 66297069 354511959 186513433 943129396 803520305 80483720 256829546 503170069 860699487 510068918 119545819 798398751 535770994 175117886 678046917 152509878 732985457 10648887 584040673 320035663 253820867 16801600 623490591 787558808 682105001 508754902 39448982 615806514 720642288 470760232 176823683 965216148 567666128 569390594 343817397 625532432 842409584 126918655 
53
197425431 372608887 238587960 205686015 85262857 217567053 44689977 977387126 612526942 794539298 848448131 323330751 58860995 553364031 90862815 219465454 597674116 343853866 453407381 307623335 555762132 789558022 91842053 269948815 498289460 802272189 785187003 916443504 747793024 182568934 309084112 217432048 793874095 585822388 191645677 667245745 951818049 672902081 94785183 259753607 418622068 466651724 108851742 124484293 340870505 307095327 438260663 652854850 356137414 406475695 568667472 55696002 431902469 
61
243024869 926465399 496193230 794575806 520182724 145885541 286959054 689164654 801704036 502642902 851650709 585231589 17024841 303205871 611751882 733218535 635676509 609556779 191866370 414019607 589912003 546622530 143984455 263245260 519676127 299272241 722924109 917667362 421900185 228406297 46242643 440698133 289786440 316282550 600864729 173672465 809338665 411302821 767696843 39107888 16007331 229774138 923269647 425230670 368711259 272413351 799981329 37577702 502624350 462338919 970010489 746919544 698158330 515162035 935367156 994996655 947293630 692729740 890876336 866025052 778967490 
20
517823507 846280167 456206219 526054360 88520361 522294558 857562425 455171547 933950677 593298849 464699985 300746907 705984458 152518559 907936315 438354657 84295595 33703462 421611380 757328825 
27
764350960 36367569 34282230 266128999 383250518 347604463 318977478 363938564 163657840 415595001 407416072 216539980 63214800 21649472 956657496 793382724 219238285 67009253 182618489 289592202 781077515 408853887 760348189 771240199 490335639 101436044 838135519 
7
284089969 369048647 415116457 492815851 183433347 354278991 132342001 
92
711385729 95210560 256331874 835746321 415684668 391456024 998996144 699687521 100217545 367675649 193489459 138058562 714905607 233187926 647209307 262821737 736166029 831514859 708764806 692854263 598036593 585290205 913536289 694861263 24969699 339275041 768640281 590136743 184841059 573898011 242617843 407758611 15396959 970355013 764444245 309298310 189567240 189961250 879162349 284001397 501479905 288974409 227222589 234730475 901981510 537032371 727318588 160652606 916335694 506115722 128581810 629196471 203978235 600898291 276485894 617323859 20865005 933123040 269302279 98040789 886331942 692971604 177244139 12386842 331358596 616679201 989808938 59535199 664195282 560025441 592335214 840503855 944399143 230920049 653948841 659390105 904060096 180639582 176213844 401551160 336851372 148586358 415087788 697669478 627085724 190996676 351640384 161888776 566893936 533813955 909754408 93236325 
17
439398055 289526945 279411408 929461939 798373523 876107838 749494787 538450228 12516005 351264679 224356649 634296930 220605424 405426098 124191281 28704418 95884691 
38
802596968 942621039 726387632 630433341 268439450 837514809 257021580 693860430 750092829 320655375 561859904 788176987 743588204 79568135 693230452 513300643 97496587 357193453 715279108 569502148 377540379 885748342 61328941 277538131 351278587 158405655 923757586 161596296 214537068 26817617 410129898 281506705 403781795 869453502 874132074 521609823 88210681 724164459 
66
197356320 700594010 78914647 963873196 539435336 190592470 610830452 384717035 203048326 334574052 476977838 448053424 246396634 353422667 694213295 281066287 111610888 919431529 42032417 337897128 84044004 351116608 782452247 954461034 317111110 477601760 344098998 65796434 234797665 814850579 98719536 609132611 514326144 377109403 131991492 566083423 90115262 993943899 95398445 886790709 400709655 268316271 708559126 693351015 278009986 428658790 196844318 795533340 73134953 911265306 329500297 510352426 273542734 368224616 762323005 601592166 449816646 168766743 683112929 670671715 160991870 903234905 321384234 146083368 79783009 406589271 
90
644413087 497393962 224829362 147437004 130548225 726313221 426848391 891796598 268058018 321879426 20721574 734292702 371360465 284330167 668338013 878625847 575990337 822950667 793405392 274068328 308249601 920397489 732199161 280155073 489164620 775714753 530585775 388787510 844715472 511389580 729443182 404014497 402610187 135016903 685666562 851869887 164445258 744966154 280314346 417582815 70327905 380505216 510707188 917351495 395827385 276850095 269873477 801275588 425656104 902470156 606308777 448769888 451938659 713523425 975187101 897173223 894581458 297637155 722481009 492832014 866795214 113380358 824826133 302161022 880260302 63563497 824570996 56674146 221125092 288096546 256180475 280572884 963986962 364454585 20965863 721777290 200564472 324358775 797410555 973121040 363993242 931327027 787365690 734736621 668525378 884485519 218616773 750819243 570592362 866343869 
73
722535732 734053404 53502775 117433126 313999207 816851499 340081856 161737134 970235685 936629732 408075056 928907604 289334589 729695136 461950318 792819340 773820841 343982701 140090327 300303127 856272400 519445425 768355510 721670759 10475547 299774656 445662584 681683890 115137239 393458278 376114212 309412441 459105707 871964407 696925037 304419489 227090679 93183386 176035445 838293585 280183687 166237437 59751998 282853862 81425072 62125769 344125297 202957443 139297823 948665301 597576704 594547293 505085981 161455915 827745594 253651095 156168956 38494174 242215251 759367480 134632597 112999341 435048116 331870914 831383435 238229389 404018489 93001446 236415900 207077403 928593575 160329752 85618565 
10
167674825 497432890 882933673 891289604 110962392 16895500 46344303 291092605 664469747 16095720 
15
348909289 890164253 512127768 36435571 304727060 146708542 300667433 666440870 233040078 870014325 597136525 45078871 568049028 367993402 371888444 
6
974931949 608245637 455239592 55049903 332697535 701823191 
47
334510011 928711315 101574038 551599391 943602256 368881368 391785814 941860658 756308876 838059733 396371800 884410063 907750764 521600159 649164214 492576296 176321887 297301404 22442648 100354240 898236624 141981730 646476483 536842144 683914581 407399289 423746991 246898848 141632103 807061121 109172451 848747331 531182510 251820702 125213609 194469745 805566212 173634255 302034281 235928724 690346662 676908472 25996923 968296706 217678098 244380112 338133641 
30
800045275 219300957 760921605 714629155 308569076 830806774 898766555 125884264 457647093 840337966 614402055 452092020 497739149 360476887 49301492 647157990 319833358 759981154 62261476 370596944 674772099 383192414 297239945 926137404 277655292 307971687 846380864 873870178 922468044 632069813 
73
311323091 393787819 313347581 70533258 479495188 834984687 712057828 260831854 618557641 364537538 143460063 701942865 694715494 324348115 674663129 671483631 782053607 954114917 296799775 900522326 36062380 473513314 27392056 408783268 170524068 124443449 419378141 81407400 192015318 523476030 787667315 101875142 448568775 577832559 23824274 828950073 605688405 209961564 301720314 224521211 897088485 18779051 372071148 907251039 583459218 50412953 301429614 634793666 260719237 249249716 595465836 334577586 742730977 159326664 845017557 439984806 683411776 299875869 679449076 444506791 636088911 640517531 308729711 751076166 415738485 38720859 119428883 326235622 881788111 90532340 674245851 510542071 162747728 
65
990478118 405585386 156327602 349899282 930797987 810005629 301808078 914662002 266925645 691738821 364792767 55885260 709804908 313336631 491961394 315538342 257345882 885231162 168361031 605775517 295256207 788567868 363883728 592248793 838680373 302838318 968632153 215720395 482487025 815963270 708282588 132535846 802174953 609266495 241288551 713382924 136182858 369236396 910444344 588103565 742857354 270598615 421008476 151009372 633162628 731343187 956114651 717625152 550355623 289083671 247524614 484365807 928429222 293833980 399888495 309796556 291940802 47959556 31765730 789900493 531985950 415136162 433929867 456184880 223130538 
7
805888823 934142346 863274382 805945098 511464621 361149378 373307906 
86
357079960 166185460 959570335 842662984 85905585 998266459 538683717 277418116 890926073 58079204 117429575 373371789 11655952 122917817 494606260 573521471 589398308 923132963 709442211 208763476 741194384 681336492 513016846 422168965 27993145 443140596 505726948 823860964 163293019 860384535 836336692 130249868 74031754 888370435 227463342 843726226 396824693 880382302 761903279 69792160 849451129 751788593 250802830 184039353 315986675 911758856 291058503 611051458 559733765 923490907 202028895 96873125 104636812 75883457 246752379 935774792 821733089 868936059 317238600 962849505 93242523 941518392 579353732 125733214 812039898 253104504 946908694 724485029 682018941 571326528 183155766 874507064 165086333 402513927 697639224 445154966 912485252 796430995 640315021 393910897 88178570 634422598 135815408 582580054 395136197 495493036 
4
656481562 666116346 100417755 94455393 
68
981366040 64923627 940366265 660167892 298672845 106405508 29454818 803485213 812921611 17833528 213686205 609580557 569394167 808557285 674710777 767363725 671233103 449268996 142520585 68573627 322302775 327903815 752099902 192543160 688053790 986118597 399527681 12317471 934435115 853188405 355782390 966191213 217381627 207609819 270574924 202182403 555229608 288572209 576141983 403837014 691856411 824018625 554271402 76738031 526225315 208185380 510738290 80923481 993821235 432556027 557166333 659691317 178608957 201329145 958393277 954509381 85580903 995472017 698245495 153924403 177463439 764141208 88896480 650291052 390843983 608260438 29475698 629352739 
76
71140176 323812712 821301965 687280636 263839355 861446762 338639757 767466483 924260257 742605887 376597721 651015875 413919503 825839696 210910616 954250044 727839174 828528834 498219364 97710980 156996584 505892340 982083796 861119486 402243009 500774748 157586346 120145515 735887742 157929629 552056397 867731175 798832636 355864745 267385811 261730007 566836163 522004832 396156361 713740505 25161713 502919227 696686439 339824889 51301868 82416790 388980642 856955198 784001770 249271507 380882295 123602440 674406935 522937579 944011177 531515582 316648386 912699684 46039287 966834661 643644451 849658821 728299118 821170853 162973172 349402758 591913264 479676242 584664305 879495573 511338252 930164892 76779695 408712407 616985614 329381507 
72
281417613 609501434 665897120 221586390 977936806 577180514 176355595 468855001 28172877 542615998 55833333 942039780 848650724 190377817 720987608 56921897 26987341 226527985 633141215 413216680 537054043 214075652 78516790 574003501 844816861 717871246 866595355 751712122 801767820 108867295 204338681 413969207 476216746 307833327 612156614 919458851 266401102 825788244 24386035 345336588 434959739 50234890 78777285 998767085 775956579 128946362 231378283 780399053 570958432 322337481 129531811 156624197 195569537 652443074 256126023 850607506 919363829 154788223 468181786 407011695 362035022 424371012 11179479 847091554 404571011 741265471 94683522 876548610 740169778 16163450 748745312 473756617 
25
924526952 694455428 250453554 647173253 357737258 914050735 684180613 678474349 433940260 405781306 943817018 268788475 292958743 392796030 200285413 48221723 236853712 396714630 772914099 316140623 520911044 203315248 87912847 292381949 516847896 
34
250642809 726229134 777541022 581678856 930959670 555398470 478447834 958193446 733631524 610206943 304861323 460135320 667683466 674746526 319213998 978495289 499939587 469327320 497978369 988245084 665691848 68241865 81099924 30193592 347187326 727291583 780612337 479220217 183308572 606712212 50970657 252933849 842423077 766345770 
33
253334091 333419452 267878091 952701437 653535556 431123593 50381014 402048083 241868650 578109424 778895404 112499240 366051003 936874623 969052006 346161077 161101641 820629077 854424816 276162935 39588416 550074044 611368570 330460825 496809736 907244953 624677612 549752572 761163224 591546616 70200285 19935334 251694237 
97
840339459 189784247 705608077 168223437 603511839 305340921 117853480 62073188 207866581 162434506 305109791 178654583 942496298 979905537 747809066 600236360 288605161 841588902 770308794 303219237 221060800 130436078 729958017 803655473 176401547 267317460 787884437 101970426 466190931 643885076 17535223 346349226 534581792 293168928 255741711 78932717 151001063 680914726 18667986 565342478 847260944 735454393 820971167 655341818 105968865 968683888 614487958 29512130 434773830 233299612 849580258 903205252 864187958 229611978 84008954 14278353 896499488 678202109 202366974 481650861 842033007 966216096 836280350 655634064 208551737 775393298 311175281 261002539 115271325 976587947 308782203 35167277 773371377 818856071 359869428 643136659 77446958 599435653 899011257 798111527 186053828 99538021 907547560 191462717 486453742 74711570 389956320 987737384 238519637 180361971 202025003 929762686 567346084 453759489 427138391 618372705 221147687 
36
527550459 456903023 219128713 75573778 931163137 477887277 241717761 46324494 883900400 742484329 541424143 139499083 842454878 436638993 356853996 44224554 210838688 213418308 858458766 834115034 153539782 340988512 311161641 19295289 613777137 527755647 777017476 824390373 752760296 745943883 230880293 502511760 409915712 836399985 883571620 531126926 
52
177437566 150450876 395319869 646573909 143373350 498380440 136403377 110791222 111966039 661646549 360001472 870761256 958146883 644522322 768738253 153283181 451719167 393283893 976378321 932853343 507372576 700083392 151318169 905717511 737319098 773757339 685176215 348645810 194619231 976571732 462438554 969561848 79154433 854159744 603026747 279313149 183515089 955761878 908147435 287927542 778652169 20764599 988064220 742922846 208515592 897494469 382486671 223478436 494849306 950776822 117465291 633558324 
65
114430272 398232169 646308434 338631221 472729240 417378736 490445432 44262392 351674354 687990876 138914095 50525406 736434559 613846000 111336223 244141527 594024445 874739549 23203201 62859687 611048808 248629592 229392906 850172948 56129998 924681328 248882259 686010149 288287020 719091166 933051921 50110641 169089599 726960183 353640723 440466012 630844659 710627158 111032216 554551248 948612785 611492133 129051415 978032690 762270419 563982067 405802063 700672521 576386283 269416739 868034528 519112682 442095893 922854140 335830318 497000973 940416178 358520853 539036231 25829634 773077016 238238952 86673382 781712239 560633896 
18
569405220 421787260 243385717 308120416 203095188 283265102 734786824 386204221 111150797 640641262 281653890 714416964 44568217 303215102 842728037 541381062 583427062 785380299 
31
852817311 878000048 964341602 676847341 844287583 860073424 704960163 948622881 551392769 141771413 723771686 95784769 16649120 632497381 396692227 453926961 613608108 807696515 952479768 450540404 717284918 825108450 955874688 690322281 682703412 677446315 598749469 328966780 760153550 299466888 942739063 
12
131569835 292515455 628201931 282982538 650747356 570652698 329087535 415370865 362093855 769058542 632101005 243075809 
63
108545214 806907862 807308075 738940228 522928785 481687501 544127339 339515841 64900365 289876263 112644285 83946690 956909430 510055861 901358239 220080616 385019442 176205329 295312764 662811486 505093019 679186674 898288109 288809376 267194271 486896746 708244991 739319772 431906068 284212931 986323023 466457839 144581184 220478515 30717769 294135425 824556142 568095733 993158406 69632123 590163709 956400457 963185465 119479429 438001655 101745831 226941838 245264486 97572056 780316987 793143542 293336702 63383173 951747496 522549767 300572754 84240995 251826723 906849275 67978636 994969526 723272077 990140866 
94
330298359 70135434 147818490 484265214 146297396 369881922 425810218 846232032 398719816 753391253 814515162 922992632 86335155 282374641 366974251 71081480 283171437 233186457 300641129 444792710 879099078 331446207 305538859 91116802 61451476 757777392 914052297 61442657 358214627 520130829 415102360 645494312 629186376 983734902 897862969 577243789 74588778 588994407 669827637 526904708 225199738 317000432 604176085 901296902 320992045 922181872 126597628 92211467 996581390 359668139 364096529 330677601 645159803 484332525 283010005 581515022 73837611 251377459 890067205 259138750 819412428 799316591 702054660 932391506 762730434 212628975 941482032 31316170 354396878 311252348 324870295 195099214 175456855 370402026 278773201 154280643 811644724 987830445 694540485 758069062 776605638 319325508 580526310 435556827 727710014 867157401 89131428 788530166 117850362 169450437 556738655 728626755 345568354 600878439 
21
995774278 475140097 688245531 381458716 997618154 712138682 325570479 313924275 724665333 60508995 672576448 412849637 440724510 874869083 446576234 44008247 224618702 142970077 650618676 806831137 11372472 
67
56582214 867647285 743380159 30019367 804497536 213223110 893248992 893012354 97135287 493780781 369017193 320140619 244385572 634205423 473828611 382341475 277759318 696486231 172085738 70157949 326450016 709701638 815940771 282416515 583567210 437966593 762851214 957225243 454167900 39704540 209979217 780150966 931895029 112811161 726128859 157209761 107441540 882843712 611513394 800994422 274902169 988839555 290132188 447770930 237948529 536717701 100930051 640339391 985222881 471105801 374931587 201351340 622773531 999862250 72453051 475279323 167210190 888123351 369496062 333435516 366971382 729968866 151017086 434761609 350451779 58937515 125228829 
38
101766045 310318889 865915775 39988029 450500710 486347054 512116610 375510814 83912918 32333957 122569399 624564389 265609706 425939371 143264867 415376053 103159506 688214039 713033291 495609796 228805532 220930140 507750727 982090591 149559334 14936647 515363972 670224029 733860315 716391900 726486702 675624716 913000319 961292276 52452419 95223990 222982482 182695791 
37
150055760 193291771 194997264 119091201 294457605 19789589 710657980 303778743 130520886 262081310 655558121 24347946 136090446 325946068 772032151 177965945 246332376 278154794 132917917 520421462 170767690 134159379 122930295 656987816 72136511 24087981 654458307 837903397 176034103 76786304 61887704 719519069 525376560 789520318 237161946 336861817 876761482 
64
349872439 779974968 374876130 518779563 737397230 562491733 298075207 807945190 401747906 558645755 334420707 39891197 509874572 138172077 319453741 234269429 224836819 669733452 20184950 732869469 852840629 78604451 272865904 303534319 209194580 620110456 595178392 602827139 276107516 109634063 772947654 90935180 186588400 43170364 274740078 206080387 263870321 982348324 251270028 867252084 583398396 595274607 608783720 965240811 509695908 390555710 347644942 194120166 537989528 161068473 228898229 656396473 89168707 830529179 147885553 402400544 128144602 171502912 985497451 532505397 234355471 392885835 714353675 879009265 
8
564227052 550610613 239156224 27631111 756468689 802419974 678333061 556370005 
91
905105578 582334399 944272226 583583331 797205135 886216065 46708454 326370057 117847630 739603614 449714381 551596770 998206441 576483010 229106457 37816886 265960124 502452583 164913760 98977970 700163510 892872016 220374847 528592036 811812158 54702507 504048342 902301396 262657216 946022945 453612464 733247093 621287974 104505853 265416683 507862683 770232845 683383370 630673634 315107440 259414162 244955563 859803844 610391192 660204288 612557969 162181479 321067462 97374908 644845103 272823041 704355861 778438768 722471261 40800512 15028276 998193770 974191659 782206475 427687291 453940230 411240347 616283120 450039803 127953566 431283222 225951414 115162343 185653801 579364402 253389620 393467099 264412693 18442454 669327685 583499400 949577792 929917446 639812539 847135459 186928063 655026938 12473086 54616608 176654789 547404522 415650614 683735146 482959343 26820262 270859288 
14
465477118 970691004 891785451 743431049 24457452 16709306 162230448 824056955 56230457 594174544 175098204 663003700 206903076 708855144 
31
49369316 768386932 227683870 310971969 217628025 152421612 453599538 299166784 625287012 821965753 209126417 410789553 252936721 975384866 609350188 336801684 553023397 519098618 558389279 48599912 695232210 801012082 595191972 846638042 665888401 767590690 562088031 334886387 875660360 702567921 474453064 
48
25518369 888125093 204807837 909854126 334120161 171125891 147503494 685360693 947961112 149305323 199698393 84185605 516017596 507289255 793339116 342082803 501266301 140541617 234379872 327605082 111182397 567644288 430975580 459551967 68340629 814223964 744048363 216946247 651797710 17230326 577831185 732586876 275902732 401538655 582472168 746102152 669389717 204907886 334326015 975591026 829346257 123743273 904266914 631673784 283739990 655213273 192329733 711637564 
19
235856019 100225000 220395495 58525778 404952943 834317791 886081783 679703492 389085530 709279744 35865395 504895409 496508385 55223653 514670367 736862027 692208277 503757721 638756352 
63
558111193 916654564 883468474 259106113 912327603 942789991 590985817 921614762 358361823 836735271 33643362 415497282 344719380 370491102 163795667 719564447 113792014 823539042 449278534 174760653 625435989 219159602 641482644 353268490 727697149 655938046 452061316 748508672 601250342 285751161 708834799 217213178 72609502 764452600 316869685 942683359 643237226 282775833 926175906 835699019 740693639 626382937 91950891 30749615 614252772 199559518 717905571 305543551 137421562 577219056 957719442 783074900 160680526 282706594 328987633 455685848 495499131 924857543 53127308 643083563 280526663 937754334 960594611 
75
42005742 264450465 796749630 614072714 846395654 387468860 423966477 700601303 764828595 749050782 166047729 39671782 884591211 97092045 501745232 78260739 176427486 415495841 805069467 347079183 344080428 366752662 681981751 465842459 296467982 46582991 94494077 77748277 961927812 990320092 419335204 446245807 242713088 217264760 804190138 988065283 59120163 564906621 40230826 886158232 866902730 288492105 393402017 561079465 622648267 286756946 434452144 533554715 804117864 298154755 101875190 524300310 173186546 922017146 50028861 444164887 61223703 796108057 416437701 275434840 140407562 757960186 429217109 119590246 399938043 151448127 821260075 303938569 855058244 218814917 673922595 159406682 144541179 320948228 17987755 
45
994411764 240901747 443970979 873787383 482807212 372504052 61810696 153390240 742748696 276621001 226440386 770769956 112660045 48738198 103665972 259876470 191068707 374984983 863849016 190390261 675815703 184606565 930934538 835151956 19489218 332544803 341055152 226487440 553768007 402043335 438571875 459191182 236213397 969313305 347236831 901952860 411965386 509433052 980759628 104877067 90294699 108645551 218578170 548237227 904082485 
100
977714960 929558698 827825269 269731315 322289668 964421018 206020268 482665970 692269115 396215440 702340468 219480230 839569473 655079351 941152563 239522037 129252191 18167340 730793658 925307078 483424878 454440550 475104623 200910598 684767359 863347922 518958123 23721586 359688286 393803115 608100493 347886535 620851570 624880318 77449079 169404380 278187848 774637815 995267555 216125829 161021890 338309363 530401902 24146631 112577377 529040186 284656814 649643032 642618811 170535308 299051200 897044772 407006937 572310622 38699467 183753096 614713756 148030685 771699757 558409127 33289123 833918407 959020087 934356651 814270403 948681032 879994748 155823193 114291505 958713487 549716189 502249726 506550896 300849525 909815724 621250224 911690102 870399114 550302822 119565418 109739422 38875809 211160360 919684293 96073654 825120681 907672578 670325843 355780940 529610527 896072523 528599990 809205681 235922724 287554767 45813687 303205058 566820944 284230135 132171750 
56
375409023 314828581 444820517 35598481 978630652 282502916 281734972 430840584 921261241 222225965 750347586 732757030 559969788 860668118 72257883 437531704 312659460 52392808 389907372 146605782 686642232 995698121 608696676 150466397 588600446 611392563 115853598 490159614 483356409 643956366 422628342 415751948 445260018 959868402 499127983 843067313 717742112 944073064 875509157 211161027 145814853 686815784 146965885 472053096 367402392 605938007 131538111 769738644 372245831 343631431 342550207 856277787 634253992 665231462 26397290 359367200 
19
418413099 209911625 285553864 19230067 411825901 195825866 393966065 328701674 139144311 228974167 784370371 446121873 85846118 213147657 619425656 116520364 545060132 168105157 258859269 
69
244163409 96926612 320313303 682967723 718106062 800530855 65682555 779469018 110613065 761523214 206918224 85623870 498546819 885776771 330883417 669159251 532160978 668640082 244023005 726657395 71981312 212622159 968410426 849698962 484356504 473627951 265499448 147169775 334335732 425097028 573731934 255963243 115614548 859147757 641358922 125283634 965341617 312257245 610907927 712883286 348674174 607248202 745493481 542829716 971953353 333631511 571823821 556799978 104473594 163527421 440660877 155986343 736615333 93159869 146852593 777448765 157856381 219373759 433844024 225366958 42012216 310655943 958353138 802150612 609644491 624397061 533950516 29401100 460682859 
31
191397340 103449020 880933086 389602922 181514923 253371252 504231742 481096026 676097652 653060793 926397640 427703533 961876212 872476180 156682179 131316504 178385021 84293369 104490174 364125814 894492712 777353303 754215800 691762970 201272811 616601643 124132798 71349517 231660686 344147190 736341309 
60
243312863 433240709 436178028 80940743 508748708 465646560 143171179 336762954 33523220 314767664 459186208 206791071 548466699 440547255 668241712 920956035 506947393 577364751 524191314 251300004 891868336 546775361 975305114 622968923 637001771 184099512 522711410 566399857 800839353 272409739 705615052 735847602 185296923 44985460 93127915 130892975 245564995 234446068 265038620 927017569 325198863 495993525 521295254 503980509 601295869 409487964 365050596 50289530 900244290 700974281 282339785 182824188 981580273 951931469 956551457 330987457 51382151 690164395 468025064 736442918 
91
373321205 835425055 215222015 204512619 316334112 136229488 206544914 79969008 912801645 403930164 802885133 726216211 883539477 253993804 26599610 741750722 500245612 262337306 647380165 370758155 592925358 973732183 676637163 336632981 667861019 692211257 436309531 842564893 645349565 628175645 641383235 394309930 144218530 489736278 741479673 518810081 941067675 593806915 674567779 648658203 95594180 781774348 25799496 520983706 726350612 858291905 32614679 943991005 262476616 880662038 996523253 471864483 485795451 639616528 208037777 348324920 998952383 435621477 402235402 432767425 731436333 299886542 654536760 411598476 648542252 897173067 185027268 668703916 449648353 475184054 309986176 366071520 426716271 470018591 876986380 978629959 953617786 879414853 851461680 118833385 777199538 66886506 380651473 341770544 281401877 583295990 362191864 707298739 601715999 298106637 558851990 
43
397832751 15441877 325690717 335811552 421809237 960434455 243150691 978661082 841137630 270585405 707256818 122479217 998706031 21792733 466562633 360504687 855179583 301035538 125022595 546283615 626296522 522020188 615131591 742834938 338910922 568274534 920110995 978195203 891663802 39469827 525519895 359779983 719479902 178295615 874231958 129812868 362858215 13145283 480503908 509136470 498381975 365623862 656602656 
99
350695787 247415602 653859192 444659925 429403367 257043952 364621636 140447136 296422375 166104524 237452376 260010514 970770771 69165283 710568453 780705260 529604264 422525567 636020518 944410012 316917552 533004995 152105235 732103963 66307597 867540348 272697055 278086073 234119037 980047485 386702237 358641209 518232093 488566078 737062156 273742985 425342676 643180127 621655793 765347515 672650435 834440049 127789130 801048696 371926287 94532978 570789252 94172585 585248260 870462336 80024001 227961144 514155197 266819212 771508569 324758727 935875119 219970926 238622401 417006738 203691252 272985688 497452640 198882310 516570291 621019959 886331378 636910326 538179032 204577422 363589511 831966749 291100709 108898014 147665102 566517406 293319547 22476423 615469275 41015426 202929663 763481604 863067024 57365931 805981198 112438408 924950185 257821789 654342548 816586198 625721473 801137599 82322968 915517440 391961466 201346011 418841597 144396167 329741331 
21
48731821 241195748 764691725 75586153 811536900 889910136 395702913 994312930 642210829 130923817 865727696 542190718 893033733 415818444 140429896 439529008 616633650 175970125 235190841 135784451 874927793 
87
711765245 343759098 579815548 685738587 644222055 683034878 320853977 707630248 28958226 70136694 444886794 63344404 220160533 863889513 508736410 515418104 128204627 320986515 791550982 783944678 631184951 954616331 453938385 721146994 447870985 129248725 727424610 607999714 87623096 378240960 875598719 343803718 145950765 363581482 224177533 373282045 222458865 892887637 852142461 877251164 428878314 993802789 674180477 160031941 764768521 74185961 831818951 517832268 795016432 385153788 240084163 307971709 603480188 951343446 90467441 817754871 171884173 611274638 870108463 449477091 749435632 378137795 48261630 751285397 263772152 512138621 525720669 122573101 71866891 575088521 219924239 891294297 845469101 29836251 113018290 809247012 188658520 445672996 512186752 45852408 42005535 148772499 711451993 221178252 926228999 930457464 475352003 
67
186906360 393970873 327299019 327980235 756474224 971516376 265623277 301422044 63718925 928059321 946945973 692505528 378961504 980112328 13884084 892955658 732293602 625946942 137045998 159026420 414832539 212056150 223463122 317786099 143526714 685914832 119841737 177713651 343354200 23123076 140430219 143692374 368598620 561020134 292242624 878172695 215743968 185171063 483093465 47011483 212281980 613193298 930659655 246973556 962196793 148037042 545142770 52549543 832614330 37036516 609020620 331993834 287096831 27373474 213768510 842782923 339046392 995939537 594446523 664952587 800823852 479610666 948159688 846063471 301486479 419861111 164096907 
57
600561116 571795169 742764602 646119331 215400870 922732606 684893505 964863342 144279442 655371201 636255258 895364833 318764260 260041315 781259332 942858498 51394706 233730685 494167448 877842220 659824902 880182520 727735137 499665151 955439679 35678654 839412687 529950069 622543064 882693281 820104830 138390623 963982827 256074650 151352786 148760884 52915305 483411616 865565133 297969161 166423902 694579335 448269239 179302539 635717053 728955917 625523036 106906754 375249773 726752944 112490491 503947667 364948108 782948098 390693094 424821950 29976082 
