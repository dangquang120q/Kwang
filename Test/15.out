-1623352406
-1371336289
-1608589296
-1713985530
-1336513250
-1338676715
-1449318617
-1382666522
-1595517879
-1706791557
1276980955
-1335807201
-1691023996
-1542840329
-1382689167
-1382672193
-1439436578
-1856274272
-1868347848
-2009966519
-1441207676
-1547595293
-1533079014
-1379742374
-1691605470
-1374280997
1423015663
-1319555447
-1391798795
-1376223625
-1512572591
-1370033477
-1436339230
-1350736428
-1596332139
-1353953023
-1427905643
-1932072136
-1422271238
2051906903
-1316698498
-1326820559
-1426705781
-1321042610
-1438584110
-1790782099
-1361880710
-2057745572
-1324375426
-1608433886
-1597284028
-1461561032
-1837705695
-1433863252
-1354654109
-1350482599
-1357563763
-1360770121
1948856159
-1389261900
-1512200732
-1381530452
-1320861701
-1339404980
-1399739028
-1517710497
-1395204730
-1347399055
-1410681448
